sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOverrides.scala

```diff
@@ -3014,6 +3014,48 @@ object GpuOverrides extends Logging {
 
         override val supportBufferConversion: Boolean = true
       }),
+    expr[ApproximatePercentile](
+      "Approximate percentile",
+      ExprChecks.aggNotReduction(
+        // note that output can be single number or array depending on whether percentiles param
+        // is a single number or an array
+        TypeSig.gpuNumeric + TypeSig.ARRAY.nested(TypeSig.gpuNumeric),
+        TypeSig.numeric + TypeSig.DATE + TypeSig.TIMESTAMP + TypeSig.ARRAY.nested(
+          TypeSig.numeric + TypeSig.DATE + TypeSig.TIMESTAMP),
+        Seq(
+          ParamCheck("input", TypeSig.gpuNumeric, TypeSig.numeric + TypeSig.DATE +
+            TypeSig.TIMESTAMP),
+          ParamCheck("percentage", TypeSig.DOUBLE + TypeSig.ARRAY.nested(TypeSig.DOUBLE),
+            TypeSig.DOUBLE + TypeSig.ARRAY.nested(TypeSig.DOUBLE)),
+          ParamCheck("accuracy", TypeSig.INT, TypeSig.INT))),
+      (c, conf, p, r) => new TypedImperativeAggExprMeta[ApproximatePercentile](c, conf, p, r) {
+
+        override def tagExprForGpu(): Unit = {
+          if (!conf.isApproxPercentileEnabled) {
+            willNotWorkOnGpu(s"approx_percentile on GPU is disabled. " +
+              s"Set ${RapidsConf.ENABLE_APPROX_PERCENTILE}=true to enable it")
+          } else if (!childExprs.tail.forall(_.wrapped match {
+            case lit: Literal => lit.value != null
+            case _ => false
+          })) {
+            willNotWorkOnGpu("approx_percentile on GPU only supports non-null literal " +
+              "expressions for percentiles and accuracy")
```

Comment:
What about an array with nulls in it?

Reply:
We now fall back to CPUI for arrays containing nulls and a test has been added.

Reply:
Do we need a follow-on issue to support arrays with nulls on GPUs?

Reply:
It is just a really odd corner case. I am find with it as is right now.